<?xml version="1.0"?>
<launch>
  <arg name="fixed_frame" default="openni_depth_frame" />

  <arg name="debug" value="False" />
  <arg if="$(arg debug)" name="launch_prefix" value="xterm -e gdb --args" />
  <arg unless="$(arg debug)" name="launch_prefix" value=""/>

  <param name="/use_sim_time" value="False" />
  <arg name="tf_prefix" value="/skeleton" />
  

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher">
    <param name="publish_frequency" value="20.0"/>
  </node>
    
  <include file="$(find openni_launch)/launch/kinect_frames.launch" />

  <group if="$(arg debug)">
    <node launch-prefix="$(arg launch_prefix)" name="skeleton_tracker" pkg="skeleton_markers" type="skeleton_tracker" output="screen"	>
      <param name="fixed_frame" value="$(arg fixed_frame)" />
      <param name="load_filepath" value="$(find skeleton_markers)/params/SamplesConfig.xml" />
    </node>
  </group>
  <group unless="$(arg debug)">
    <node name="skeleton_tracker" pkg="skeleton_markers" type="skeleton_tracker">
      <param name="fixed_frame" value="$(arg fixed_frame)" />
      <param name="load_filepath" value="$(find skeleton_markers)/params/SamplesConfig.xml" />
    </node>
  </group>
</launch>
